### ISSUE TITLE
[Bug]: operator restart node  in loop

### ISSUE THREAD

**Issue by oudb** [strimzi/strimzi-kafka-operator#10253]
### Bug Description

we change chang num.io.threads=4，operator will Dynamic update, kafka cr will be ready.

then we chang num.io.threads=14、offsets.retention.minutes=10080，becase offsets.retention.minutes is readonly,operator will  restart kafka node， kafka cr will be ready， num.io.threads is stiil 4.

and wait about STRIMZI_FULL_RECONCILIATION_INTERVAL_MS， operator redo reconciliation, try to change num.io.threads=14，but broker return InvalidRequestException，operator will restart kafka node.

I think :
1、operator should  do dynamic update after restart kafka node in the first reconciliation，this can tell we some invalid config  in time
2、operator do not restart kafka node if broker resurn error InvalidRequestException, just log errro and chang status to notready

### Steps to reproduce

1.create single-node cluster，and wait ready
kubectl apply -f https://strimzi.io/examples/latest/kafka/kraft/kafka-single-node.yaml -n kafka 
kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka 

2、chang num.io.threads=4, and wait ready， this will Dynamic update
kubectl patch kafka/my-cluster -n kafka --type='json' -p="[{\"op\": \"replace\", \"path\": \"/spec/kafka/config/num.io.threads\", \"value\":4}]"
kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka 

3、chang num.io.threads=14、offsets.retention.minutes=10080 and wait ready， this will restart node 0 in the first reconciliation
kubectl patch kafka/my-cluster -n kafka --type='json' -p="[{\"op\": \"replace\", \"path\": \"/spec/kafka/config/num.io.threads\", \"value\":14},{\"op\": \"replace\", \"path\": \"/spec/kafka/config/offsets.retention.minutes\", \"value\":10080}]"
kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka 

4、wait about STRIMZI_FULL_RECONCILIATION_INTERVAL_MS，you will see log and kafka node restart：

2024-06-24 07:38:00 ERROR KafkaRoller:744 - Reconciliation #159(timer) Kafka(kafka/my-cluster): Error updating broker configuration for pod my-cluster-dual-role-0/0
org.apache.kafka.common.errors.InvalidRequestException: Invalid value org.apache.kafka.common.config.ConfigException: Dynamic thread count update validation failed for num.io.threads=4, value should be at least half the current value 14 for configuration Invalid dynamic configuration

### Strimzi version

0.41.0

### Kubernetes version

Kubernetes 1.25

**Comment by oudb** [comment]
[op.log](https://github.com/user-attachments/files/15951686/op.log)
the operator log file

**Comment by scholzj** [comment]
@oudb I guess this is a duplicate of #7391 which is caused by how Kafka does some special validation for some fields and how it misdesigned the dynamic update functionality? I do not think it has really any way to fix it in Strimzi.

(Keep in mind that the change from `num.io.threads=14` to `num.io.threads=4` that you requested was invalid)

**Comment by oudb** [comment]
@scholzj  if requested was invalid， oprator  should change status to not ready，but status is ready, some automated tool  make some error op.

**Comment by scholzj** [comment]
That is easy to say, but there is no way how to validate it.

**Comment by oudb** [comment]
I change num.io.threads=14、offsets.retention.minutes=10080, and then do: 
kubectl wait kafka/my-cluster --for=condition=Ready --timeout=300s -n kafka

kubectl tell me condition are met, but some error in my request，and i do not get it in time

**Comment by oudb** [comment]
if i submit a modify that need  restart and a modify that had invald config，in the first reconciliation, opearor tell me that is ok

**Comment by scholzj** [comment]
I guess I do not understand what exactly are you complaining about. Sorry. Maybe you should provide a clearer example with what you expect to happen and what happens. Because I do not think this it is clear.

The scenario you provided is indeed problematic and does not work well - but it is already documented in #7391 where a change to the fields with hidden internal update logic leads to a conflict between static configuration file and dynamically updated configuration in Apache Kafka that causes infinite rolling of the Kafka nodes.

**Comment by oudb** [comment]
If modify kafka cr，how i know  modification is done and modification is ok or error？

**Comment by scholzj** [comment]
The operator works in the reconciliation loop. If it sees some need for a action to adopt the change, it will take the action. The modification here is ok, so the operator assumes the limitation was addressed ... until Kafka reverts it back due to how its dynamic update system is designed. So the operator does not really know about it not succeeding and cannot report it. 

Again, this is from my point of view described in the pre-existing issue I linked.
